File: ABC01_99/ABC43/c.cpp
#include <algorithm>
#include <cmath>
#include <functional>
#include <iostream>
using namespace std;

int main() {
  int N;
  cin >> N;
  int a[N];
  for (int i = 0; i < N; i++) {
    cin >> a[i];
  }

  int ans = 1000000000;
  for (int i = -100; i <= 100; i++) {
    int sum = 0;
    for (int j = 0; j < N; j++) {
      sum += (a[j] - i) * (a[j] - i);
    }
    ans = min(sum, ans);
  }

  cout << ans << endl;
  return 0;
}

File: ABC01_99/ABC43/c.py
N = int(input())
a = list(map(int,input().split()))
a.sort()
ans = 10**9
for y in range(a[0], a[-1]+1):
  sum = 0
  for i in range(N):
    x = a[i]
    sum += (x-y)**2
  ans = min(ans, sum)
  if(ans == 0): break
print(ans)